plugin/metrics/prometheus/factory.go

```diff
@@ -66,3 +67,29 @@ func (f *Factory) Initialize(logger *zap.Logger) error {
 func (f *Factory) CreateMetricsReader() (metricsstore.Reader, error) {
 	return prometheusstore.NewMetricsReader(f.options.Configuration, f.logger, f.tracer)
 }
+
+func (f *Factory) configureFromOptions(o *Options) {
+	f.options = o
+}
+
+func NewFactoryWithConfig(
+	cfg config.Configuration,
+	logger *zap.Logger,
+) (*Factory, error) {
+	if err := cfg.Validate(); err != nil {
+		return nil, err
+	}
+
+	defaultConfig := DefaultConfig()
+	cfg.ApplyDefaults(&defaultConfig)
```

Comment:
I don't think this is needed. In v1 initialization I believe defaults are supplied through default values of the CLI flags. And for v2 initialization you have custom Unmarshal function that gets default config and then overlays YAML config into it. So this ApplyDefault should never be needed.